### DISCUSSION TITLE
[Posible Bug]

### DISCUSSION THREAD

**Discussion by Alexdnk** [lllyasviel/Fooocus#2675] [General]
Maybe I'm wrong, but the **perform_upscale** function doesn't need to be called for **Fast 2x** upscale (it's Lanczos 2x)? This function in this mode only increases the time at Fast 2x upscale. 
https://github.com/lllyasviel/Fooocus/blob/e2f9bcb11d06216d6800676c48d8d74d6fd77a4b/modules/async_worker.py#L542
```python
        if 'upscale' in goals:
            H, W, C = uov_input_image.shape
            progressbar(async_task, 13, f'Upscaling image from {str((H, W))} ...')
            uov_input_image = perform_upscale(uov_input_image)
            print(f'Image upscaled.')

            if '1.5x' in uov_method:
                f = 1.5
            elif '2x' in uov_method:
                f = 2.0
            else:
                f = 1.0
```

**Comment by mashb1t** [comment]
This is intended, images are upscaled in these steps:

1. ESRGAN (upscaling x4)
2.  LANCZOS (downscaling to target upscale factor)
3. (only non-fast) SD generation

Upscaling with ESRGAN first, then downscaling again provides a much clearer picture compared to just upscaling with LANCZOS.

<details><summary>Example</summary>
<p>

original (also SD generated)
![image - 2024-01-15T154732 713](https://github.com/lllyasviel/Fooocus/assets/9307310/32063c79-42d6-4006-b75e-7236a0d9b911)

only_lanczos
![only_lanczos](https://github.com/lllyasviel/Fooocus/assets/9307310/3123685a-c1f3-4863-8c9f-f67f1d85d69a)

esrgan_and_lanczos
![esrgan_and_lanczos](https://github.com/lllyasviel/Fooocus/assets/9307310/90ea4794-55f3-4c33-af7b-f226076f3b39)

</p>
</details>

**Reply by Alexdnk** [reply]
Thanks, I got it.
